### PR TITLE
capz: add presubmit conformance job to k/k

### PR DESCRIPTION
**Pull request by chewong**
Signed-off-by: Ernest Wong <chuwon@microsoft.com>

Adding `pull-kubernetes-e2e-capz-conformance` to test https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1264.

/assign @CecileRobertMichon

### CODE REVIEW

File: config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml [hunk 1]
```diff
@@ -214,6 +214,42 @@ presubmits:
...
+        - ./scripts/ci-conformance.sh
+        securityContext:
+          privileged: true
+        resources:
+          requests:
+            cpu: 1
+            memory: "4Gi"
+        env:
+        - name: KUBETEST_CONF_PATH
+          value: /home/prow/go/src/sigs.k8s.io/cluster-api-provider-azure/test/e2e/data/kubetest/conformance-fast.yaml
```
Comment (chewong) [inline config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml:247]: Want to make sure the test duration is less than an hour

Comment (CecileRobertMichon) [reply config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml:247]: should we set the timeout to less than 3 hours if that's the goal?

Comment (chewong) [reply config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml:245]: removed the timeout (the default is 2h IIRC)

File: config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml [hunk 2]
```diff
@@ -214,6 +214,40 @@ presubmits:
...
       testgrid-alert-email: kubernetes-provider-azure@googlegroups.com
       testgrid-num-columns-recent: '30'
+  - name: pull-kubernetes-e2e-capz-conformance
+    decorate: true
+    always_run: false
+    optional: true
+    labels:
+      preset-dind-enabled: "true"
+      preset-kind-volume-mounts: "true"
+      preset-azure-cred: "true"
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml:224]: here you need 

```suggestion
            preset-azure-cred-only: "true"
           preset-azure-anonymous-pull: "true"
  ```
  
  to be able to use the capzci registry

Comment (chewong) [reply config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml:224]: Good catch

Comment (CecileRobertMichon) [reply config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml:224]: did you push an update? I don't see it fixed

Comment (chewong) [reply config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml]: pushed

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

**Pull request merged**